core/src/PrognosticData.cpp

```diff
@@ -81,18 +81,15 @@ void PrognosticData::update(const TimestepTime& tst)
     pOcnBdy->updateBefore(tst);
     pAtmBdy->update(tst);
 
-    // Fill the values of the true ice and snow thicknesses.
-    iceGrowth.initializeThicknesses();
-    // Fill the updated ice temperature array
-    ticeUpd.data().setData(m_tice);
```

Comment:
Were those not needed after all?

Reply:
These two calls were just to fill arrays the the dynamics will read and change. The thermodynamics fills these arrays and so they do not need explicitly filling by hand in `PrognosticData::update()`.

`iceGrowth.initializeThicknesses()` is the first thing called by `IceGrowth::update()` (`iceGrowth.cpp` L 147), so it only needed to be called before updating the dynamics.

The updated ice temperature array `ticeUpd` is the one where the thermodynamic changes to the ice growth are written (but not read), so it doesn't need filling before the dynamics, either.

Reply:
Nice. That makes this change even more elegant ;)